src/traits.rs

```diff
@@ -8,6 +8,19 @@ pub trait Parameterized {
     fn emit_params(&self) -> Self::Parameters;
 
     fn from_params(params: Self::Parameters) -> Self;
+
+    // TODO: If Self: Sized ok here? Should this be a requirement for the whole trait?
```

Comment:
I think in every case the parameters should be sized.

Reply:
Ok I changed it to 
```rust

pub trait Parameterized: Sized {
    type Parameters;

    fn emit_params(&self) -> Self::Parameters;

    fn from_params(params: Self::Parameters) -> Self;

    fn map_params(
        &self,
        f: impl Fn(Self::Parameters) -> Self::Parameters,
    ) -> Self {
        let params = self.emit_params();
        let new_params = f(params);
        Self::from_params(new_params)
    }
}
```